packages/lib/server/getLuckyUser.ts

```diff
@@ -272,68 +203,338 @@ async function filterUsersBasedOnWeights<
 
   // Find users with the highest booking shortfall
   const maxShortfall = Math.max(...usersWithBookingShortfalls.map((user) => user.bookingShortfall));
-  const usersWithMaxShortfall = usersWithBookingShortfalls.filter(
-    (user) => user.bookingShortfall === maxShortfall
-  );
+  const usersWithMaxShortfall = usersWithBookingShortfalls.filter((user) => {
+    console.log("user.bookingShortfall", user.bookingShortfall, maxShortfall);
+    return user.bookingShortfall === maxShortfall;
+  });
 
   // ff more user's were found, find users with highest weights
   const maxWeight = Math.max(...usersWithMaxShortfall.map((user) => user.weight ?? 100));
 
   const userIdsWithMaxShortfallAndWeight = new Set(
-    usersWithMaxShortfall.filter((user) => user.weight === maxWeight).map((user) => user.id)
+    usersWithMaxShortfall
+      .filter((user) => {
+        const weight = user.weight ?? 100;
```

Comment:
Defaults user.weight as well to 100 as we are defaulting to it at other places too.

It was causing an error when no weights were defined